samples/ambient-argo/istio/cni.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: istio-cni
+  namespace: argocd
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  project: default
+  destination:
+    name: ambient-cluster
+    namespace: kube-system
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+    syncOptions:
+      - CreateNamespace=true
+  sources:
+    - repoURL: 'https://istio-release.storage.googleapis.com/charts'
+      targetRevision: 1.18.5
```

Comment:
```
      targetRevision: 1.18.5
```

How will this be updated? Can it be templated?

Reply:
This is updated when the user wishes to upgrade their cni to a new version, usually around the same time they promote rapid to stable.  While it might be possible to template it, it's already a helm chart (template) being called from an Argo App (template) which is installed by another argo app (template).  It can't be templates all the way down, and I want to make sure this remains accessible to users.

Reply:
Yep that's fair, I just want to understand which file has to be maintained - existing `bookinfo` has a maintenance shell script to update N number of references in N files, and I would prefer *one* file with *one* set of references to update so we don't

Reply:
Is that for updating Istio versions, or the versions of bookinfo images?  I think I could add the applications folder to your bookinfo updater if you can point me to it...